src/Directive.php

```diff
@@ -29,10 +32,27 @@ abstract class Directive
     public const WAKE_LOCK = 'wake-lock';
     public const XR = 'vr';
 
-    public static function isValid(string $directive): bool
+    public static function make(string $directive, string $type = 'default'): DirectiveContract
+    {
+        return match ($type) {
+            'default' => DefaultDirective::directive($directive),
+            default => throw InvalidDirective::unknownDirectiveType($type),
+        };
+    }
+
+    public function addRule(string $rule): static
```

Comment:
Introduced a method to add a rule for a directive, rule will be added to the `rules` property.